cli/cmd/helm/upgrade.go

```diff
@@ -179,3 +188,187 @@ func upgradeTobs(cmd *cobra.Command, args []string) error {
 	fmt.Printf("Successfully upgraded %s.\n", root.HelmReleaseName)
 	return nil
 }
+
+func UpgradePathBasedOnVersion(deployedChartVersion, newChartVersion string) error {
+	nVersion, err := utils.ParseVersion(newChartVersion, 3)
+	if err != nil {
+		return fmt.Errorf("failed to parse latest helm chart version %w", err)
+	}
+
+	dVersion, err := utils.ParseVersion(deployedChartVersion, 3)
+	if err != nil {
+		return fmt.Errorf("failed to parse deployed helm chart version %w", err)
+	}
+
+	fmt.Println(deployedChartVersion, nVersion, dVersion)
+	switch {
+	case deployedChartVersion <= "0.2.2" && nVersion > dVersion:
```

Comment:
these comparisons aren't right, you can't use string comparison. also I think you want nVersion > cutVersion not nVersion > dVsersion
```suggestion
	cutVersion, err := utils.ParseVersion("0.2.2", 3)
	if err != nil {
		return fmt.Errorf("failed to parse deployed helm chart version %w", err)
	}
	fmt.Println(deployedChartVersion, nVersion, dVersion)
	switch {
	case dVersion <= cutVersion && nVersion > cutVersion:
```

Reply:
So here the condition is
`dVersion <= version0_2_2 && nVersion > dVersion` so I expect the deployed version to be <= 0.2.2 and new Version to be greater than deployed Version.`

The deployed version in users env can be 0`.1.1` too.